sdk/common/src/main/java/io/opentelemetry/sdk/internal/IncludeExcludePredicate.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.internal;
+
+import static io.opentelemetry.sdk.internal.GlobUtil.toGlobPatternPredicate;
+import static java.util.stream.Collectors.joining;
+
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.StringJoiner;
+import java.util.function.Predicate;
+import javax.annotation.Nullable;
+
+/**
+ * A predicate that evaluates if a string matches a configurable set of {@code included} and {@code
+ * excluded} string.
+ *
+ * <p>Supports optional glob pattern matching. See {@link GlobUtil}.
+ *
+ * <p>String equality is evaluated using {@link String#equalsIgnoreCase(String)}.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ */
+public final class IncludeExcludePredicate implements Predicate<String> {
```

Comment:
We have a recurring pattern of needing to match strings against a configurable allow / deny list, with optional glob pattern matching. In all these cases, its beneficial to have a useful `toString()` implementation.

This PR refactors the places this pattern occurs with a common implementation.